spec/requests/api/v1/keywords_controller_spec.rb

```diff
@@ -1,20 +1,50 @@
 # frozen_string_literal: true
 
+require 'byebug'
+require 'rails_helper'
+
 RSpec.describe Api::V1::KeywordsController, type: :request do
   describe 'GET#index' do
-    it 'returns success status' do
-      get api_v1_keywords_path, headers: create_token_header
+    context 'given a logged in user gets theirs keyword' do
+      it 'returns success status' do
+        user = Fabricate(:user)
+        Fabricate.times(3, :keyword, user_id: user.id)
+
+        get api_v1_keywords_path, headers: create_token_header(user)
+
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns three items' do
+        user = Fabricate(:user)
+        Fabricate.times(3, :keyword, user_id: user.id)
+
+        get api_v1_keywords_path, headers: create_token_header(user)
 
-      expect(response).to have_http_status(:success)
+        response_body = JSON.parse(response.body, symbolize_names: true)
+        expect(response_body[:data].count).to eq 3
+      end
     end
 
-    it 'returns the valid keywords' do
-      get api_v1_keywords_path, headers: create_token_header
+    context 'given database contains keywords are not belonging to the current user' do
+      it 'returns success status' do
+        user = Fabricate(:user)
+        Fabricate.times(3, :keyword, user_id: user.id)
+
+        get api_v1_keywords_path, headers: create_token_header
+
+        expect(response).to have_http_status(:success)
+      end
 
-      keywords = JSON.parse(response.body, symbolize_names: true)[:data]
+      it 'returns an empty array' do
+        user = Fabricate(:user)
+        Fabricate.times(3, :keyword, user_id: user.id)
```

Comment:
It took a bit time to understand this 😅 , please declare things explicitly. And for this case, we also don't need to `fabricate` 3 items, it could save us some query. 

```suggestion
        another_user = Fabricate(:user)
        Fabricate.times(:keyword, user_id: another_user.id)
        
        get api_v1_keywords_path, headers: create_token_header
        
        response_body = JSON.parse(response.body, symbolize_names: true)
        expect(response_body[:data].count).to eq 0
```

Reply:
It should be?
```
        Fabricate(:keyword, user_id: another_user.id)
```

Reply:
yup, my typo issue.